### PR TITLE
feat: newsletter unsubscription

### LINKED ISSUE #652: [FEAT] Newsletter Unsubscription Enpoint
# Description:
Develop a backend endpoint to handle requests for unsubscribing from the newsletter. The endpoint will accept an email address, validate and sanitize it, and then handle both successful and failed validations appropriately. If the email passes validation and exists in the database, it will be removed, and a 'success' response will be sent to the client. If the email does not exist, an error message will be returned.

## Acceptance Criteria:
- Accepts and validates the user's email.

- Removes the data from the database if it exists.

- Returns the appropriate response and status code.

### * Requirements:
 - Implement API endpoint with data validation and sanitization.

 - Set up database integration to remove the email securely.

 - Return appropriate success or error messages.

### * Expected Outcome:
The user input data should be successfully removed from the backend, or the user will receive an error message if unsuccessful.
Status code:

- 200: Email was successfully removed.

- 400: Invalid email address.

- 404: Email not found.

- 500: A server error occurred.

### Endpoint:
[**POST**] `/api/v1/pages/newsletter/unsubscribe`

content-type: application/json

Body:

```
  {
    "email": "string"
  }
```
Responses:

  - Successful response:

    ```
      {
        "message": "Unsubscription successful.",
        "success": true,
        "status_code": 200
      }
    ```
  - Error response:

    ```
      {
        "message": "string",
        "success": false,
        "status_code": int
      }
    ```

Testing:
The system should use pytest covering:

 * Validates user's email and checks for its existence.

 * Ensures data is removed correctly from the database.

 * Tests that the appropriate response and status code is returned.

### PR DESCRIPTION
**Pull request by CaptainAril**
# Pull Request: Endpoint for unsubscribing from newsletter

## Description
This pull request implements a new API endpoint for unsubscribing from the newsletter. The endpoint will accept an email address, validate and sanitize it, and then handle both successful and failed validations appropriately. If the email passes validation and exists in the database, it will be removed, and a 'success' response will be sent to the client. If the email does not exist, an error message will be returned

## Changes
- Added  route and services for a POST endpoint - `/api/v1/pages/newsletter/unsubscribe`
- Added delete functionality for newsletter subscription
- Added comprehensive test coverage for the new endpoint

### File: `/api/v1/routes/newsletter`

```python
@newsletter.post('/pages/newsletter/unsubscribe')
async def unsubscribe(request: EmailSchema, db: Session = Depends(get_db)):
    NewsletterService.delete(db=db,  request=request)
    
    return success_response(
        status_code=200,
        message="You have unsubscribed from our newsletter."
    )

```

## Related Issue
Closes #652

### CODE REVIEW

**Comment by joboy-dev** [comment]
Update this PR
The name and the content is confusing
Which are you doing, blog or newsletter??

**Review by joboy-dev — APPROVED** [review]

File: tests/v1/newsletter/newsletter_test.py
Comment (joboy-dev) [inline tests/v1/newsletter/newsletter_test.py:1]: What happened here

File: tests/v1/newsletter/newsletter_test.py [hunk 1]
```diff
@@ -62,73 +56,11 @@ def test_sub_newsletter_existing_email(db_session_mock):
...
-
-    @classmethod
-    def teardown_class(cls):
-        app.dependency_overrides = {}
-
-    # Successfully retrieves all subscriptions from db
-    def test_retrieve_subscription_success(self, mocker):
-        mock_subs= [{"email": "test@example.com"},
-                    {"email": "test1@example.com"}]
-        mocker.patch.object(NewsletterService, 'fetch_all', return_value=mock_subs)
```
Comment (Goketech) [inline tests/v1/newsletter/newsletter_test.py:85]: you removed the tests in this file, revert changes

**Review by joboy-dev — CHANGES_REQUESTED** [review]
What happened to that test file you just removed

**Review by Goketech — CHANGES_REQUESTED** [review]

**Pull request closed without merging**